### DISCUSSION TITLE
Is it possible to incrementally add corpus to a retriever?

### DISCUSSION THREAD

**Discussion by ANYMS-A** [xhluca/bm25s#75] [Q&A] [answered]
Hi, 
currently I have a lot of corpus which I dont want to load them all into the memory and use
```python
retriever.save("bm25s_very_big_index", corpus=corpus)
```
to create the retriever.

Is it possible to incrementally add corpus to a retriever? Thanks!

**Comment by xhluca** [comment, marked as answer]
This has been discussed here: https://github.com/xhluca/bm25s/discussions/20
